templates/cluster-template.yaml

```diff
@@ -179,10 +182,13 @@ metadata:
 spec:
   template:
     spec:
+      identity: UserAssigned
       osDisk:
         diskSizeGB: 128
         osType: Linux
       sshPublicKey: ${AZURE_SSH_PUBLIC_KEY_B64:=""}
+      userAssignedIdentities:
+      - providerID: /subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${CI_RG:=capz-ci}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${USER_IDENTITY:=cloud-provider-user-identity}
```

Comment:
this is modifying the base flavor templates which will require users to pre-create the user assigned ID in their subscription prior to using them (including in the quickstart.

It's also defaulting the rg to "capz-ci" or `CI_RG` which doesn't really apply in non-CI scenarios.